mesonbuild/interpreter/compiler.py

```diff
@@ -659,7 +659,7 @@ def get_supported_arguments_method(self, args: T.Tuple[T.List[str]], kwargs: 'Ge
             if not self._has_argument_impl([arg]):
                 msg = f'Compiler for {self.compiler.get_display_language()} does not support "{arg}"'
                 if checked == 'warn':
-                    mlog.warning(msg)
+                    mlog.warning(msg, fatal=False)
```

Comment:
This seems like something that should be a fatal warning?

Reply:
oops, you're right. I misread that as "compiler does not support *checking* for arg", I'll change that back.